Move FairModule::fAllSensitiveVolumes to FairRunSim

ChristianTackeGSI opened the pull request:
Instead of using a thread local, let's use something that is tied to the "simulaion session" (as noted in some places).

FairRunSim has all the Modules anyway, so let it also have the list of sensitive Volumes.

OR should we move it to `FairMCApplcation` instead?

waiting for:
- [x] #1555

---

Checklist:

* [X] Followed the [Contributing Guidelines](https://github.com/FairRootGroup/FairRoot/blob/dev/CONTRIBUTING.md)

ChristianTackeGSI commented on fairroot/base/sim/FairMCApplication.cxx:
```diff
@@ -886,7 +886,7 @@ void FairMCApplication::InitGeometry()
 
     // Fill sensitive volumes in fVolMap
-    for (auto fv : FairModule::fAllSensitiveVolumes) {
+    for (auto fv : fRun->fAllSensitiveVolumes) {
```
The order of initialization functions is a long standing issue. This is known.

Doing checks is good idea. Just we don't know, if there might be valid cases, where `fAllSenstivieVolumes` is empty anyway.

ChristianTackeGSI commented on fairroot/base/sim/FairModule.cxx:
```diff
@@ -260,7 +259,7 @@ void FairModule::AddSensitiveVolume(TGeoVolume* v)
         volume->setModId(fModId);
         volume->SetModule(this);
-        fAllSensitiveVolumes.push_back(volume);
+        FairRunSim::Instance()->fAllSensitiveVolumes.push_back(volume);
```
Same as above.

karabowi commented:
@ChristianTackeGSI, can you add an entry in the changelog? For some users that change may be worth mentioning.
I have tested the PR with PandaRoot.

ChristianTackeGSI commented:
> @ChristianTackeGSI, can you add an entry in the changelog? For some users that change may be worth mentioning. I have tested the PR with PandaRoot.

`fAllSenstiveVolumes` was introduced during a rename for 19.0. So the rename was a breaking change. You correctly identified, that we did not document it properly. So let's document it now: #1555

After the docs have been reviewed and merged (and backported to 19.0), I will rebase this one.

ChristianTackeGSI commented:
> > @ChristianTackeGSI, can you add an entry in the changelog? For some users that change may be worth mentioning. I have tested the PR with PandaRoot.
> 
> `fAllSenstiveVolumes` was introduced during a rename for 19.0. So the rename was a breaking change. You correctly identified, that we did not document it properly. So let's document it now: #1555
> 
> After the docs have been reviewed and merged (and backported to 19.0), I will rebase this one.

- [X] #1555 has been merged
- [X] rebased
- [X] rewritten with "internal" public methods that would allow testing.
